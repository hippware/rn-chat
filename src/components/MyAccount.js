import React from 'react';
import {View, TouchableOpacity, Text, StyleSheet, ScrollView, Image} from 'react-native';
import BackgroundGradient from './BackgroundGradient';
import {k} from './Global';
import Card from './Card';
import Cell from './Cell';
import Header from './Header';
import SignUpAvatar from './SignUpAvatar';
import Separator from './Separator';
import {Actions} from 'react-native-router-native';
import {GiftedForm, GiftedFormManager} from 'react-native-gifted-form';
import validators from './FormValidators';
import LogoutButton from './LogoutButton';
import ProfileInfo from './ProfileInfo';
import Screen from './Screen';
import profileStore from '../store/profileStore';
import location from '../store/locationStore';
import model from '../model/model';
import {observer} from 'mobx-react/native';
import autobind from 'autobind-decorator';
import {colors} from '../constants';
import NavTitle from './NavTitle';
import NavBarRightButton from './NavBarRightButton';
import NavBar from './NavBar';

@autobind
@observer
export default class MyAccount extends React.Component {
  async save() {
    await profileStore.update(GiftedFormManager.stores.form.values);
    Actions.pop();
  }

  render() {
    const Group = GiftedForm.GroupWidget;
    const isDay = location.isDay;
    const profile = model.profile;
    if (!profile) {
      return <Screen isDay={isDay} />;
    }
    const {handle, firstName, lastName, email, avatar} = profile;
    return (
      <Screen isDay={isDay}>
        <GiftedForm
            testID='myAccount'
            formName='myAccount'
            formStyles={{containerView: {backgroundColor: 'transparent', paddingTop: 70 * k}}}
            validators={validators}
            defaults={{handle, firstName, lastName, email}}
        >
          <SignUpAvatar
              avatar={avatar}
              profile={this.props.profile}
              isDay={isDay}
              style={{
                top: 5,
                backgroundColor: 'rgb(243,244,246)',
                borderRadius: 33 * k,
                width: 66 * k,
                height: 66 * k,
              }}
          />

          {profile.error &&
            <Text style={{color: 'red', padding: 10, textAlign: 'center'}}>
              {profile.error}
            </Text>}

          <ProfileInfo isDay={isDay} profile={profile} editMode />

          <View style={{height: 100}}>
            <LogoutButton />

          </View>
        </GiftedForm>
        <NavBar>
          <NavTitle>@{profile.handle}</NavTitle>
          <NavBarRightButton onPress={this.save} active>
            <Text style={styles.follow}>Save</Text>
          </NavBarRightButton>
        </NavBar>
      </Screen>
    );
    // <Card isDay={isDay} style={{opacity:0.95}}>
    //   <Header>Settings</Header>
    //   <Separator width={1}/>
    //   <Cell image={require('../../images/iconVisibility.png')}>Visible to friends</Cell>
    //   <Separator width={1}/>
    //   <Cell image={require('../../images/iconLocation.png')}>Nearby filter is 2 miles</Cell>
    // </Card>
    // <Card isDay={isDay} style={{opacity:0.95}}>
    // <Header>Notifications</Header>
    // <Separator isDay={isDay} width={1}/>
    //   <Cell image={require('../../images/iconMessageXs.png')}>Notify me instantly for message activity</Cell>
    //   <Separator width={1}/>
    //   <Cell image={require('../../images/iconBotXs.png')}>Notify me for all bot activity</Cell>
    //   <Separator width={1}/>
    //   <Cell image={require('../../images/iconNotifications.png')}>Donâ€™t notify me for 3 people</Cell>
    //   </Card>
  }
}

const styles = StyleSheet.create({
  container: {
    position: 'absolute',
    top: 0,
    left: 0,
    bottom: 0,
    right: 0,
  },
  top: {
    position: 'absolute',
    top: 0,
    left: 0,
    height: 222,
    right: 0,
    opacity: 0.79,
  },
  follow: {
    color: colors.PINK,
    fontFamily: 'Roboto-Regular',
    fontSize: 15 * k,
  },
});
