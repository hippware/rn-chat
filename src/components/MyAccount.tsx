import React from 'react'
import {TouchableOpacity, View, StyleSheet, Linking, Alert} from 'react-native'
import {observer, inject} from 'mobx-react/native'
import {observable} from 'mobx'
import {k} from './Global'
import SignUpAvatar from './SignUpAvatar'
import {Actions} from 'react-native-router-flux'
import * as log from '../utils/log'
import Card from './Card'
import Cell from './Cell'
import FormTextInput from './FormTextInput'
import {colors} from '../constants'
import {RText, Separator} from './common'
import {ValidatableProfile} from '../utils/formValidation'
import {KeyboardAwareScrollView} from 'react-native-keyboard-aware-scroll-view'
import {IWocky} from 'wocky-client'
import Screen from './Screen'
import {settings} from '../globals'
import {phoneFormat} from '../utils/misc'
const {version} = require('../../package.json')

type Props = {
  wocky?: IWocky
  profileValidationStore?: any
}

@inject('wocky', 'profileValidationStore')
@observer
class MyAccount extends React.Component<Props> {
  static renderTitle = () => <Title />
  static rightButton = () => <Right />

  static submit = async profileValidationStore => {
    try {
      await profileValidationStore.save()
      Actions.pop()
      // need to force refresh of ProfileDetails after this change
      setTimeout(() => Actions.refresh({refresh: new Date()}))
    } catch (e) {
      alert(e)
    }
  }

  @observable saving: boolean = false
  @observable vProfile?: ValidatableProfile
  handle: any
  firstName: any
  lastName: any
  email: any

  componentDidMount() {
    if (this.props.wocky!.profile) {
      this.vProfile = new ValidatableProfile(this.props.wocky!.profile!)
      this.props.profileValidationStore.setProfile(this.vProfile)
    }
  }

  render() {
    const {wocky} = this.props
    const {profile} = wocky!
    if (!profile || !this.vProfile) {
      log.log('NULL PROFILE', {level: log.levels.ERROR})
      return <View style={{flex: 1, backgroundColor: 'white'}} />
    }
    return (
      <Screen>
        <KeyboardAwareScrollView testID="myAccountScrollView">
          <View style={styles.headerOuter}>
            <SignUpAvatar
              style={{
                width: 100,
                height: 100,
                borderRadius: 50,
              }}
              showDot
            />
          </View>

          <Card style={{opacity: 0.95}}>
            <RText
              size={16}
              weight="Medium"
              style={{padding: 15, color: colors.navBarTextColorDay}}
              testID="profileInfo"
            >
              About
            </RText>
            <Separator />
            <FormTextInput
              label="First Name"
              store={this.vProfile && this.vProfile.firstName}
              icon={require('../../images/iconSubsNew.png')}
              onSubmitEditing={() => this.lastName.focus()}
            />
            <FormTextInput
              ref={r => (this.lastName = r)}
              label="Last Name"
              store={this.vProfile && this.vProfile.lastName}
              onSubmitEditing={() => this.handle.focus()}
            />
            <FormTextInput
              ref={r => (this.handle = r)}
              label="Username"
              store={this.vProfile && this.vProfile.handle}
              autoCapitalize="none"
              icon={require('../../images/iconUsernameNew.png')}
              onSubmitEditing={() => this.email.focus()}
            />
            <FormTextInput
              label="Phone"
              icon={require('../../images/phone.png')}
              editable={false}
              value={phoneFormat(profile.phoneNumber!)}
              // value={format({phone: profile.phoneNumber.replace('+', ''), country: 'US'}, 'E.164')}
            />
            <FormTextInput
              ref={r => (this.email = r)}
              label="Email"
              store={this.vProfile && this.vProfile.email}
              icon={require('../../images/iconEmailNew.png')}
              onSubmitEditing={() => MyAccount.submit(this.props.profileValidationStore)}
            />
            <Cell
              image={require('../../images/blocked.png')}
              onPress={Actions.blocked}
              imageStyle={{height: 20 * k, width: 20 * k, marginHorizontal: 5 * k}}
              style={{marginTop: 0}}
            >
              <RText numberOfLines={1} size={18} style={{flex: 1, color: colors.DARK_PURPLE}}>
                Blocked Users
              </RText>
            </Cell>
            {/* <Cell image={icon} style={{justifyContent: 'center'}} imageStyle={{height: 20 * k, width: 20 * k, marginHorizontal: 5 * k}}> */}
          </Card>

          {/* <LogoutButton /> */}

          <View style={{marginVertical: 30, alignItems: 'center'}}>
            <Version />
            <LinkButton onPress={() => Linking.openURL('https://tinyrobot.com/terms-of-service/')}>
              Terms of Service
            </LinkButton>
            <LinkButton onPress={() => Linking.openURL('https://tinyrobot.com/privacy-policy/')}>
              Privacy Policy
            </LinkButton>
            <LinkButton
              onPress={() => {
                Alert.alert('Log Out', `Are you sure you want to log out?`, [
                  {text: 'Cancel', style: 'cancel'},
                  {
                    text: 'Log Out',
                    style: 'destructive',
                    onPress: Actions.logout,
                  },
                ])
              }}
            >
              Logout
            </LinkButton>
            {settings.isStaging && (
              <LinkButton
                style={{marginTop: 50}}
                onPress={() => {
                  Alert.alert(
                    'Delete Profile',
                    `Are you reeeeally sure you want to remove this profile? This action cannot be reversed.`,
                    [
                      {text: 'Cancel', style: 'cancel'},
                      {
                        text: 'Delete Profile',
                        style: 'destructive',
                        onPress: async () => {
                          Actions.logout()
                          wocky!.remove()
                        },
                      },
                    ]
                  )
                }}
              >
                Delete Profile
              </LinkButton>
            )}
          </View>
        </KeyboardAwareScrollView>
      </Screen>
    )
  }
}

export default MyAccount

const LinkButton = ({
  children,
  onPress,
  style,
}: {
  children: any
  onPress: () => void
  style?: any
}) => (
  <TouchableOpacity onPress={onPress}>
    <RText size={15} color={colors.PINK} style={[styles.text, style]}>
      {children}
    </RText>
  </TouchableOpacity>
)

const Title = inject('wocky')(
  observer(
    ({wocky}) =>
      wocky.profile ? (
        <RText
          size={16}
          style={{
            letterSpacing: 0.5,
            color: colors.DARK_PURPLE,
          }}
        >
          {`@${wocky.profile.handle}`}
        </RText>
      ) : null
  )
)

const Right = inject('profileValidationStore', 'wocky')(
  observer(({profileValidationStore, wocky}) => {
    const {profile} = wocky
    const disabled =
      !profile ||
      profile.updating ||
      profile.uploading ||
      (profile.avatar && profile.avatar.loading)
    return profile ? (
      <TouchableOpacity
        onPress={() => MyAccount.submit(profileValidationStore)}
        disabled={disabled}
      >
        <RText
          size={16}
          style={{
            marginRight: 10 * k,
            color: disabled ? colors.GREY : colors.PINK,
            opacity: profile.updating ? 0.5 : 1,
          }}
        >
          Save
        </RText>
      </TouchableOpacity>
    ) : null
  })
)

const Version = inject()(
  observer(({}) => {
    return (
      <RText size={15} color={colors.DARK_GREY} style={{marginBottom: 15}}>
        {`Version ${version}`}
      </RText>
    )
  })
)

const styles = StyleSheet.create({
  headerOuter: {
    paddingTop: 10,
    paddingBottom: 20,
    backgroundColor: 'white',
    width: '100%',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    marginBottom: 15,
  },
})
