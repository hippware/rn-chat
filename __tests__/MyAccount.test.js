// @flow

import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import MyAccount from '../src/components/MyAccount'
import {Provider} from 'mobx-react/native'

describe('MyAccount', () => {
  // TODO fix ?
  // it('renders', () => {
  //   const wocky = {
  //     profile: {
  //       handle: 'jerkham',
  //       firstName: 'eric',
  //       lastName: 'kirkham',
  //       email: 'eric.kirkham@gmail.com',
  //       loaded: true,
  //     },
  //   }
  //   const profileValidationStore = {
  //     setProfile: () => {},
  //   }
  //   const toRender = (
  //     <Provider wocky={wocky} profileValidationStore={profileValidationStore} warn={() => {}}>
  //       <MyAccount />
  //     </Provider>
  //   )
  //   const tree = renderer.create(toRender).toJSON()
  //   expect(tree).toMatchSnapshot()
  // })
})
