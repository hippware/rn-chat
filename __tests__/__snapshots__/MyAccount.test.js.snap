// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyAccount renders 1`] = `
<View
  style={
    Array [
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      },
      undefined,
    ]
  }
>
  <BVLinearGradient
    colors={
      Array [
        "rgb(255,255,255)",
        "rgb(241,242,244)",
        "rgb(243,244,246)",
      ]
    }
    locations={
      Array [
        0,
        0.2,
        1,
      ]
    }
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "flex": 1,
        },
      ]
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      collapsable={undefined}
      hitSlop={undefined}
      isTVSelectable={true}
      nativeID={undefined}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Image
        resizeMode="cover"
        source={
          Object {
            "testUri": "../../../images/addPhoto.png",
          }
        }
        style={
          Array [
            Object {
              "borderRadius": 80,
              "height": 160,
              "width": 160,
            },
            Object {
              "backgroundColor": "rgb(243,244,246)",
              "borderRadius": 66,
              "height": 132,
              "top": 5,
              "width": 132,
            },
          ]
        }
      />
    </View>
    <View
      isDay={true}
      style={
        Array [
          Object {
            "paddingBottom": 20,
            "paddingLeft": 30,
            "paddingRight": 30,
            "paddingTop": 26,
          },
          Object {
            "opacity": 0.95,
          },
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "borderColor": "white",
              "borderRadius": 2,
              "shadowOffset": Object {
                "height": 1,
                "width": 0,
              },
              "shadowOpacity": 0.12,
              "shadowRadius": 2,
            },
            Object {
              "backgroundColor": "rgba(255,255,255,1)",
            },
            undefined,
          ]
        }
      >
        <View
          isDay={true}
          style={
            Object {
              "padding": 30,
            }
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": undefined,
                  "fontFamily": "Roboto-Medium",
                  "fontSize": 16,
                },
                Object {
                  "color": "rgb(63,50,77)",
                },
              ]
            }
          >
            Profile Info
          </Text>
        </View>
        <View
          style={
            Object {
              "backgroundColor": "rgba(155,155,155,0.15)",
              "height": 2,
            }
          }
        />
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 11,
                },
                Object {
                  "marginLeft": 80,
                  "marginTop": 10,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 30,
                },
                Object {
                  "justifyContent": "center",
                  "paddingTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "paddingLeft": 10,
                    "paddingRight": 30,
                    "width": 30,
                  },
                  Object {
                    "height": 40,
                    "marginHorizontal": 10,
                    "width": 40,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/iconSubsNew.png",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                clearButtonMode="while-editing"
                icon={
                  Object {
                    "testUri": "../../../images/iconSubsNew.png",
                  }
                }
                isDay={true}
                label="First Name"
                onChangeText={[Function]}
                placeholder="First Name"
                placeholderTextColor="rgb(212,212,212)"
                returnKeyType="next"
                store={
                  ValidateItem {
                    "errorMessage": "",
                    "isValid": undefined,
                    "key": "firstName",
                    "value": "eric",
                  }
                }
                style={
                  Object {
                    "color": "rgb(63,50,77)",
                    "flex": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 36,
                  }
                }
                underlineColorAndroid="transparent"
                value="eric"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginRight": 20,
                    "width": 44,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgb(63,50,77)",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 11,
                },
                Object {
                  "marginLeft": 80,
                  "marginTop": 10,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 30,
                },
                Object {
                  "justifyContent": "center",
                  "paddingTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <View
                style={
                  Object {
                    "width": 60,
                  }
                }
              />
              <TextInput
                allowFontScaling={true}
                clearButtonMode="while-editing"
                isDay={true}
                label="Last Name"
                onChangeText={[Function]}
                placeholder="Last Name"
                placeholderTextColor="rgb(212,212,212)"
                returnKeyType="next"
                store={
                  ValidateItem {
                    "errorMessage": "",
                    "isValid": undefined,
                    "key": "lastName",
                    "value": "kirkham",
                  }
                }
                style={
                  Object {
                    "color": "rgb(63,50,77)",
                    "flex": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 36,
                  }
                }
                underlineColorAndroid="transparent"
                value="kirkham"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginRight": 20,
                    "width": 44,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgb(63,50,77)",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 11,
                },
                Object {
                  "marginLeft": 80,
                  "marginTop": 10,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 30,
                },
                Object {
                  "justifyContent": "center",
                  "paddingTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "paddingLeft": 10,
                    "paddingRight": 30,
                    "width": 30,
                  },
                  Object {
                    "height": 40,
                    "marginHorizontal": 10,
                    "width": 40,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/iconUsernameNew.png",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                autoCapitalize="none"
                clearButtonMode="while-editing"
                icon={
                  Object {
                    "testUri": "../../../images/iconUsernameNew.png",
                  }
                }
                isDay={true}
                label="Username"
                onChangeText={[Function]}
                placeholder="Username"
                placeholderTextColor="rgb(212,212,212)"
                returnKeyType="next"
                store={
                  ValidateItem {
                    "errorMessage": "",
                    "isValid": undefined,
                    "key": "handle",
                    "value": "jerkham",
                  }
                }
                style={
                  Object {
                    "color": "rgb(63,50,77)",
                    "flex": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 36,
                  }
                }
                underlineColorAndroid="transparent"
                value="jerkham"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginRight": 20,
                    "width": 44,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgb(63,50,77)",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />
        </View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "color": "rgb(253,95,108)",
                  "fontFamily": "Roboto-Regular",
                  "fontSize": 11,
                },
                Object {
                  "marginLeft": 80,
                  "marginTop": 10,
                },
              ]
            }
          >
            
          </Text>
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "padding": 30,
                },
                Object {
                  "justifyContent": "center",
                  "paddingTop": 4,
                },
              ]
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "paddingLeft": 10,
                    "paddingRight": 30,
                    "width": 30,
                  },
                  Object {
                    "height": 40,
                    "marginHorizontal": 10,
                    "width": 40,
                  },
                ]
              }
            >
              <Image
                source={
                  Object {
                    "testUri": "../../../images/iconEmailNew.png",
                  }
                }
              />
            </View>
            <View
              style={
                Object {
                  "alignItems": "center",
                  "flex": 1,
                  "flexDirection": "row",
                  "justifyContent": "center",
                }
              }
            >
              <TextInput
                allowFontScaling={true}
                clearButtonMode="while-editing"
                icon={
                  Object {
                    "testUri": "../../../images/iconEmailNew.png",
                  }
                }
                isDay={true}
                label="Email"
                onChangeText={[Function]}
                placeholder="Email"
                placeholderTextColor="rgb(212,212,212)"
                returnKeyType="next"
                store={
                  ValidateItem {
                    "errorMessage": "",
                    "isValid": undefined,
                    "key": "email",
                    "value": "eric.kirkham@gmail.com",
                  }
                }
                style={
                  Object {
                    "color": "rgb(63,50,77)",
                    "flex": 1,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 36,
                  }
                }
                underlineColorAndroid="transparent"
                value="eric.kirkham@gmail.com"
              />
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "justifyContent": "center",
                    "marginRight": 20,
                    "width": 44,
                  }
                }
              />
            </View>
          </View>
          <View
            style={
              Object {
                "backgroundColor": "rgb(63,50,77)",
                "height": 1,
                "opacity": 0.2,
              }
            }
          />
        </View>
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "padding": 30,
              },
              Object {
                "marginTop": 20,
              },
            ]
          }
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "paddingLeft": 10,
                  "paddingRight": 30,
                  "width": 30,
                },
                Object {
                  "height": 40,
                  "marginHorizontal": 10,
                  "width": 40,
                },
              ]
            }
          >
            <Image
              source={
                Object {
                  "testUri": "../../../images/block.png",
                }
              }
            />
          </View>
          <View
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={true}
              ellipsizeMode="tail"
              numberOfLines={1}
              style={
                Array [
                  Object {
                    "color": undefined,
                    "fontFamily": "Roboto-Regular",
                    "fontSize": 18,
                  },
                  Object {
                    "color": "rgb(63,50,77)",
                    "flex": 1,
                  },
                ]
              }
            >
              Blocked Users
            </Text>
          </View>
        </View>
      </View>
    </View>
    <View
      style={
        Object {
          "height": 100,
        }
      }
    >
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "stretch",
            "backgroundColor": "rgb(254,92,108)",
            "borderRadius": 4,
            "borderWidth": 0,
            "bottom": 80,
            "flexDirection": "row",
            "height": 100,
            "justifyContent": "center",
            "left": 30,
            "marginBottom": 10,
            "opacity": 1,
            "position": "absolute",
            "right": 30,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Array [
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
                "fontSize": 18,
                "textAlign": "center",
              },
              Object {
                "color": "white",
                "fontFamily": "Roboto-Regular",
                "fontSize": 30,
              },
            ]
          }
        >
          Logout
        </Text>
      </View>
    </View>
  </View>
</View>
`;
